modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1212,6 +1224,37 @@ def from_pandas(cls, df):
             new_widths,
             dtypes=new_dtypes,
             index_cols=index_cols,
+            is_default_frame=len(unsupported_cols) > 0,
+        )
+
+    @classmethod
+    def from_arrow(cls, at):
+        partitions, unsupported_cols = cls._frame_mgr_cls.from_arrow(
+            at, return_dims=True
+        )
+        new_frame, new_lengths, new_widths = partitions
+
+        new_columns = pd.Index(data=at.column_names, dtype="O")
+        new_index = pd.RangeIndex(at.num_rows)
+        new_dtypes = pd.Series(
+            [cls._arrow_type_to_dtype(col.type) for col in at.columns],
+            index=new_columns,
+        )
+
+        if len(unsupported_cols) > 0:
+            ErrorMessage.single_warning(
```

Comment:
I'm not sure we get the desired number of messages here. Consider we have a frame with unsupported data in columns `a` and `b`. Then `df["c"] = df["a"]` would produce three warning messages (for `['a', 'b']`, `['a']`, `['a', 'b', 'c']`). Is it intentional?

Reply:
You right about the number of warnings in that case, but all of that warnings will be printed when we're constructing a new frame, and since we want to warn user about unsupported columns at the construction stage, it's not clear how we can decide from modin frame constructor when we should print a warning and when we shouldn't because we have chaining operations.

Reply:
Now I think I was wrong about the number of messages. If we warn only in `from_pandas` and `from_arrow` then we would get warnings for leaf frames only and wouldn't get them for other frames in our operations chain.

Reply:
yeah, but if we're doing all operations default to pandas, then at every query compiler function we would do 
1. `.to_pandas()`
2. `func(df)`
3. `from_pandas(res)`

so in your example we will get 3 warnings

Reply:
So I was wrong that I was wrong :)